torch/_functorch/aot_autograd.py

```diff
@@ -876,6 +880,7 @@ class AOTConfig:
     dynamic_shapes: bool = False
     aot_autograd_arg_pos_to_source : Optional[List[Source]] = None
     inference_compiler: Optional[Callable] = None
+    philox_total_offsets: PhiloxTotalOffsets = PhiloxTotalOffsets()
```

Comment:
This looks suspicious. There are a few problems:

* You generated a singleton PhiloxTotalOffsets default which is shared across all AOTConfig instances. Probably undesirable
* This is a config: you expect to set it up once and then never mutate it. But PhiloxTotalOffsets feels like it's begging to be mutated. I would feel better if you had marked that dataclass as frozen, but I haven't read far enough to know if you can or not.

Reply:
This is same as https://github.com/pytorch/pytorch/pull/97377/files#r1161744760